modules/EvseManager/Charger.cpp

```diff
@@ -1023,20 +1052,66 @@ void Charger::stop_session() {
     signal_simple_event(types::evse_manager::SessionEventEnum::SessionFinished);
 }
 
-void Charger::start_transaction() {
+bool Charger::start_transaction() {
     shared_context.stop_transaction_id_token.reset();
     shared_context.transaction_active = true;
+
+    const types::powermeter::TransactionReq req{evse_id, "", "", 0, 0, ""};
+    for (const auto& meter : r_powermeter_billing) {
+        const auto response = meter->call_start_transaction(req);
+        // If we want to start the session but the meter fail, we stop the charging since
+        // we can't bill the customer.
+        if (response.status == types::powermeter::TransactionRequestStatus::UNEXPECTED_ERROR) {
+            EVLOG_error << "Failed to start a transaction on the power meter " << response.error.value_or("");
+            set_faulted();
```

Comment:
This should not set faulted directly, it should emit an error so that this can be handled by the error framework and e.g. forwarded to OCPP etc. There is an error definition already: PowermeterTransactionStartFailed in errors/evse_manager.yaml. Please rebase on RC1 before that as the error handling in EvseManager was changed in RC1.